### ISSUE TITLE
grad-norm goes to infinity

### ISSUE THREAD

**Issue by mmahdavian** [yuantianyuan01/StreamMapNet#37]
Hello

I am trying to train your model on my own dataset. The data is similar to yours, but during training, total loss is going down, but grad-norm goes toward a very large number. Do you have any idea what the problem is?

`2024-09-10 10:11:04,258 - mmdet - INFO - Iter [100/15000]       lr: 4.653e-05, eta: 2:21:23, time: 0.569, data_time: 0.015, memory: 10185, cls: 2.0250, reg: 8.0096, d0.cls: 1.7710, d0.reg: 5.4187, d1.cls: 1.8796, d1.reg: 5.7607, d2.cls: 1.9309, d2.reg: 6.3147, d3.cls: 1.9443, d3.reg: 6.8999, d4.cls: 1.9833, d4.reg: 7.4638, trans_loss: 0.3656, total: 51.7671, grad_norm: 117.7429
2024-09-10 10:12:00,385 - mmdet - INFO - Iter [200/15000]       lr: 5.984e-05, eta: 2:19:26, time: 0.561, data_time: 0.011, memory: 10187, cls: 1.8068, reg: 3.9424, d0.cls: 1.7532, d0.reg: 3.1035, d1.cls: 1.7710, d1.reg: 3.2074, d2.cls: 1.7839, d2.reg: 3.3318, d3.cls: 1.8029, d3.reg: 3.4483, d4.cls: 1.7943, d4.reg: 3.6625, trans_loss: 0.2620, total: 31.6701, grad_norm: 79.9597
2024-09-10 10:12:56,791 - mmdet - INFO - Iter [300/15000]       lr: 7.313e-05, eta: 2:18:23, time: 0.564, data_time: 0.010, memory: 10187, cls: 1.8157, reg: 4.1265, d0.cls: 1.7351, d0.reg: 3.0604, d1.cls: 1.7727, d1.reg: 3.3041, d2.cls: 1.7666, d2.reg: 3.4419, d3.cls: 1.7788, d3.reg: 3.6435, d4.cls: 1.7911, d4.reg: 3.9292, trans_loss: 0.1339, total: 32.2995, grad_norm: 94.7493
2024-09-10 10:13:53,022 - mmdet - INFO - Iter [400/15000]       lr: 8.638e-05, eta: 2:17:18, time: 0.562, data_time: 0.010, memory: 10187, cls: 1.8036, reg: 6.9279, d0.cls: 1.8568, d0.reg: 4.8340, d1.cls: 1.7862, d1.reg: 4.5244, d2.cls: 1.7738, d2.reg: 4.6532, d3.cls: 1.7311, d3.reg: 5.1499, d4.cls: 1.6901, d4.reg: 6.6303, trans_loss: 0.0986, total: 43.4599, grad_norm: 4395.1608
2024-09-10 10:14:49,780 - mmdet - INFO - Iter [500/15000]       lr: 9.960e-05, eta: 2:16:31, time: 0.568, data_time: 0.010, memory: 10187, cls: 1.8824, reg: 4.2541, d0.cls: 2.0430, d0.reg: 6.2246, d1.cls: 1.8598, d1.reg: 5.3739, d2.cls: 1.9115, d2.reg: 5.4601, d3.cls: 1.7929, d3.reg: 5.0303, d4.cls: 1.7577, d4.reg: 4.9602, trans_loss: 0.1645, total: 42.7150, grad_norm: 82008.9710
2024-09-10 10:15:46,454 - mmdet - INFO - Iter [600/15000]       lr: 9.961e-05, eta: 2:15:39, time: 0.567, data_time: 0.010, memory: 10187, cls: 1.8818, reg: 4.6010, d0.cls: 2.0847, d0.reg: 6.3212, d1.cls: 1.7512, d1.reg: 6.4953, d2.cls: 1.8128, d2.reg: 5.1590, d3.cls: 1.8329, d3.reg: 5.4288, d4.cls: 1.7811, d4.reg: 5.2385, trans_loss: 0.1895, total: 44.5778, grad_norm: 192633.9689
2024-09-10 10:16:43,125 - mmdet - INFO - Iter [700/15000]       lr: 9.947e-05, eta: 2:14:45, time: 0.567, data_time: 0.010, memory: 10189, cls: 1.8473, reg: 3.8787, d0.cls: 2.2551, d0.reg: 5.0335, d1.cls: 1.7849, d1.reg: 4.3462, d2.cls: 1.7813, d2.reg: 4.5727, d3.cls: 1.9119, d3.reg: 3.7656, d4.cls: 1.8885, d4.reg: 3.9942, trans_loss: 0.2864, total: 37.3462, grad_norm: 1933278.2293
2024-09-10 10:17:42,181 - mmdet - INFO - Iter [800/15000]       lr: 9.930e-05, eta: 2:14:33, time: 0.591, data_time: 0.010, memory: 10189, cls: 1.8339, reg: 3.7209, d0.cls: 2.3032, d0.reg: 4.9928, d1.cls: 1.7194, d1.reg: 4.4010, d2.cls: 1.7578, d2.reg: 4.4207, d3.cls: 1.8762, d3.reg: 3.8000, d4.cls: 1.8574, d4.reg: 3.8171, trans_loss: 0.1582, total: 36.6586, grad_norm: 1677193.1403
2024-09-10 10:18:41,288 - mmdet - INFO - Iter [900/15000]       lr: 9.912e-05, eta: 2:14:12, time: 0.591, data_time: 0.010, memory: 10189, cls: 1.8651, reg: 3.9634, d0.cls: 2.4052, d0.reg: 5.7300, d1.cls: 1.7871, d1.reg: 4.6843, d2.cls: 1.8218, d2.reg: 4.1139, d3.cls: 1.8533, d3.reg: 4.0005, d4.cls: 1.8695, d4.reg: 3.9320, trans_loss: 0.1641, total: 38.1902, grad_norm: 2526629.2270`

**Comment by yuantianyuan01** [comment]
Hi, sorry for the late reply. It seems the training just failed to converge at the very beginning. I recommend try a smaller learning rate to warm up or a larger batch size.
